src/directives/click-outside/click-outside.ts

```diff
@@ -0,0 +1,80 @@
+import { DirectiveOptions } from 'vue';
+import { VNodeDirective } from 'vue/types/vnode';
+
+interface ClickOutsideBindingArgs {
+	closeConditional?: (e: Event) => boolean;
+	include?: () => HTMLElement[];
+}
+
+interface ClickOutsideDirective extends VNodeDirective {
+	value?: {
+		cb: (e: Event) => void;
+		args: ClickOutsideBindingArgs;
+	};
+	args?: ClickOutsideBindingArgs;
+}
+
+function closeConditional() {
+	return false;
+}
+
+function directive(e: PointerEvent, el: HTMLElement, binding: ClickOutsideDirective): void {
+	// Args may not always be supplied
+	binding.args = binding.value?.args || {};
+
+	// If no closeConditional was supplied assign a default
+	const isActive = binding.args.closeConditional || closeConditional;
```

Comment:
```suggestion
	const isActive = binding.args.closeConditional || false;
```

I don't think this needs a separate function